File: DataStructure/DifferentialArray/leetcode2251/differential_array_2251.cpp
// class Solution 
// {
// public:
//     vector<int> fullBloomFlowers(vector<vector<int>>& flowers, vector<int>& people) 
//     {
//         map<int,int>diff;
//         for(const auto&flower:flowers)
//         {
//             ++diff[flower[0]];
//             --diff[flower[1]+1];
//         }

//         const int n= people.size();
//         vector<int>id(n);

//         iota(id.begin(),id.end(),0);
//         sort(id.begin(),id.end(),[&](int i,int j){return people[i]<people[j];});


//         auto it=diff.begin();
//         int sum=0;
//         for(int i:id)
//         {
//             while(it!=diff.end() && it->first<=people[i])
//                 sum+=it++->second;
//             people[i]=sum;
//         }

//         return people;
//     }
// };

File: DataStructure/DifferentialArray/leetcode2251/differential_array_2251.go
// func fullBloomFlowers(flowers [][]int, people []int) []int {
//     diff:=map[int]int{}

//     for _,flower:=range flowers{
//         diff[flower[0]]++
//         diff[flower[1]+1]--
//     }

//     	n := len(diff)

// 	times := make([]int, 0, n)

// 	for t := range diff {

// 		times = append(times, t)

// 	}

// 	sort.Ints(times)

//     id:=make([]int,len(people))
//     for i:=range id{
//         id[i]=i
//     }
//     sort.Slice(id,func(i,j int)bool{return people[id[i]]<people[id[j]]})

//     j,sum:=0,0
//     for _,i := range id{
//         for ;j<n && times[j]<=people[i];j++{
//             sum+=diff[times[j]]
//         }
//         people[i] = sum
//     }
//     return people
// }